src/FieldWarning/Assets/Ingame/UI/SelectionManager.cs

```diff
@@ -233,15 +238,22 @@ private bool IsInside(Vector3 t)
             return insideX && insideY;
         }
 
-        private void UnselectAll(List<PlatoonBehaviour> l, bool justPreviewing)
+        private void UnselectAll(List<PlatoonBehaviour> selectedPlatoons, bool justPreviewing)
         {
-            l.ForEach(x => x.SetSelected(false, justPreviewing));
-            l.Clear();
+            if (selectedPlatoons != null) {
+                selectedPlatoons.ForEach(x => x.SetSelected(false, justPreviewing));
+                selectedPlatoons.Clear();
+            }
         }
 
-        private void SetSelected(List<PlatoonBehaviour> l, bool justPreviewing)
+        private void SetSelected(List<PlatoonBehaviour> selectedPlatoons, bool justPreviewing)
         {
-            l.ForEach(x => x.SetSelected(true, justPreviewing));
+            if(selectedPlatoons != null) {
+                selectedPlatoons.ForEach(x => x.SetSelected(true, justPreviewing));
+                // Randomly choose one platoon to play a selected voiceline
+                int randInt = Random.Range(0,selectedPlatoons.Count);
+                selectedPlatoons[randInt].PlaySelectionVoiceline();
+            }
```

Comment:
Lines 254-255 cause a range exception as soon as you enter the scene (you roll 0 and try to access the zeroth platoon, but the selection is empty).

A couple places in your commit also see unnecessary added null checks (line 251, 243 etc..). The problem with these is that they hide errors. The code must not, and currently does not, call these methods with null objects. If they are ever called with a null object the code should fail loudly and we should do our best to restructure it so it doesn't do that anymore. Sometimes you can't do without null checks but if we are unnecessarily defensive in our programming we leave space for errors to hide under the rug.

Otherwise, great job. I also made a minor tweak to the prefab you committed (it was missing an audio source to play the voicelines with), and merged your stuff into master.